### PR TITLE
[contrib] A finger tree implementation

### PR DESCRIPTION
**Pull request by AliasQli**
A finger tree implementation.

### CODE REVIEW

File: libs/contrib/contrib.ipkg [hunk 1]
```diff
@@ -89,6 +89,9 @@ modules = Control.ANSI,
 
           Data.Rel.Complement,
 
+          Data.Seq,
+          Data.Seq.Seq',
```
Comment (Z-snails) [inline libs/contrib/contrib.ipkg:93]: Why are there 2 of these?

File: libs/contrib/Data/Seq/Seq'.idr [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+public export
+implementation Monoid (Seq' a) where
+  neutral = empty
+
+||| This implementation is differnt from that of Seq.
+public export
+implementation Applicative Seq' where
+  pure = singleton
+  fs <*> xs = concat $ map (\f => map f xs) fs
```
Comment (Z-snails) [inline libs/contrib/Data/Seq/Seq'.idr:204]: `concat . map` = `foldMap` (unless it's not `Prelude.concat`)

File: libs/contrib/contrib.ipkg (hunk 1 above)
Comment (AliasQli) [reply libs/contrib/contrib.ipkg:93]: The first one has its length in its type, the second not.

File: libs/contrib/Data/Seq/Seq'.idr (hunk 2 above)
Comment (AliasQli) [reply libs/contrib/Data/Seq/Seq'.idr:204]: Ok, I forgot.

**Comment by edwinb** [comment]
This looks nice, although I'm now in two minds about merging this sort of contribution. In the past, we've always been fairly happy to add anything to `contrib` but we're now at a stage where there's a lot there that risks becoming unmaintained, and a lot of it is untested. So, would this be better put into its own library, which we can link from https://github.com/idris-lang/Idris2/wiki/Libraries?

My worry in general is that merging a new library also implies some kind of committment on our part to maintain it, and we already probably have too much to keep track of (as evidence of that: we're too slow at dealing with PRs!)

**Pull request merged**